cmd/cluster-init/update_jobs.go

```diff
@@ -0,0 +1,141 @@
+}
+
+type Post struct {
+	OSRelease struct {
+		Jobs []prowconfig.Postsubmit `json:"openshift/release,omitempty"`
+	} `json:"postsubmits,omitempty"`
+}
+
+type Pre struct {
+	OSRelease struct {
+		Jobs []prowconfig.Presubmit `json:"openshift/release,omitempty"`
+	} `json:"presubmits,omitempty"`
+}
+
+func updatePresubmits(o options) {
+	presubmitsFile := filepath.Join(o.releaseRepo, CiOperator, Jobs, Openshift, Release, "openshift-release-master-presubmits.yaml")
+	fmt.Printf("Updating Presubmit Jobs: %s\n", presubmitsFile)
+	presubmits := &Pre{}
+	loadConfig(presubmitsFile, presubmits)
+	presubmit := generatePresubmit(o.clusterName, o.buildFarmDir)
+	presubmits.OSRelease.Jobs = append(presubmits.OSRelease.Jobs, presubmit)
+	saveConfig(presubmitsFile, presubmits)
+}
+
+func updatePostsubmits(o options) {
+	postsubmitsFile := filepath.Join(o.releaseRepo, CiOperator, Jobs, Openshift, Release, "openshift-release-master-postsubmits.yaml")
+	fmt.Printf("Updating Postsubmit Jobs: %s\n", postsubmitsFile)
+	postsubmits := &Post{}
+	loadConfig(postsubmitsFile, postsubmits)
+	postsubmit := generatePostsubmit(o.clusterName, o.buildFarmDir)
+	postsubmits.OSRelease.Jobs = append(postsubmits.OSRelease.Jobs, postsubmit)
+	saveConfig(postsubmitsFile, *postsubmits)
+}
+
+func updateInfraPeriodics(o options) {
+	ipFile := filepath.Join(o.releaseRepo, CiOperator, Jobs, InfraPeriodicsFile)
+	fmt.Printf("Updating Periodic Jobs: %s\n", ipFile)
+	ip := &InfraPeriodics{}
+	loadConfig(ipFile, ip)
+	rotSASecretsPer, err := findPeriodic(ip, PerRotSaSecs)
+	check(err)
+	appendNewClustersConfigUpdaterToKubeconfig(rotSASecretsPer, "", o.clusterName)
+	appendBuildFarmCredentialSecret(rotSASecretsPer, o.clusterName)
+	for _, perName := range []string{PerCiSecGen, PerCiSecBoot} {
+		per, err := findPeriodic(ip, perName)
+		check(err)
+		appendNewClustersConfigUpdaterToKubeconfig(per, CiSecretBootstrap, o.clusterName)
+		appendBuildFarmCredentialSecret(per, o.clusterName)
+	}
+	ap := generatePeriodic(o.clusterName, o.buildFarmDir)
+	ip.Periodics = append(ip.Periodics, ap)
+	saveConfig(ipFile, *ip)
+}
+
+func appendNewClustersConfigUpdaterToKubeconfig(per *prowconfig.Periodic, containerName string, clusterName string) {
+	container, err := findContainer(per.Spec, containerName)
+	if err != nil {
+		logrus.WithError(err).Fatal()
+	}
+	env, err := findEnv(container, Kubeconfig)
+	check(err)
+	s := fmt.Sprintf(":/%s/%s/%s.%s.%s.%s", Etc, BuildFarmCredentials, Sa, ConfigUpdater, clusterName, Config)
+	env.Value = env.Value + s
+}
+
+func appendBuildFarmCredentialSecret(per *prowconfig.Periodic, clusterName string) {
+	v, err := findVolume(per.Spec, BuildFarmCredentials)
+	check(err)
+	configPath := fmt.Sprintf("%s.%s.%s.%s", Sa, ConfigUpdater, clusterName, Config)
+	path := v1.KeyToPath{
+		Key:  configPath,
+		Path: configPath,
+	}
+	v.Secret.Items = append(v.Secret.Items, path)
+}
+
+func findPeriodic(ip *InfraPeriodics, name string) (*prowconfig.Periodic, error) {
+	idx := -1
```

Comment:
IMHO It should be easier to create a map with the job names, and then just check if the key exists.